subsys/testsuite/ztest/src/ztest.c

```diff
@@ -416,24 +415,16 @@ int main(void)
 void main(void)
 {
 #ifdef CONFIG_USERSPACE
-	struct k_mem_partition *parts[] = {
-#ifdef Z_LIBC_PARTITION_EXISTS
-		/* C library globals, stack canary storage, etc */
-		&z_libc_partition,
-#endif
+	/* Partition containing globals tagged with ZTEST_DMEM and ZTEST_BMEM
+	 * macros.
+	 */
+	k_mem_domain_add_partition(&k_mem_domain_default,
+				   &ztest_mem_partition);
 #ifdef Z_MALLOC_PARTITION_EXISTS
-		/* Required for access to malloc arena */
-		&z_malloc_partition,
+	/* Allow access to malloc() memory */
+	k_mem_domain_add_partition(&k_mem_domain_default,
+				   &z_malloc_partition);
 #endif
-		&ztest_mem_partition
-	};
-
-	/* Ztests just have one memory domain with one partition.
-	 * Any variables that user code may reference need to go in them,
```

Comment:
The ZTEST_DMEM/BMEM remark is still valid, move that to above where the partition is added?

Reply:
There's still a comment which reads "Partition containing globals tagged with ZTEST_DMEM and ZTEST_BMEM macros" right before where ztest_mem_partition is installed.

Reply:
I added the sentence " Any variables that user code may reference need to be placed in this partition if no other memory domain configuration is made."